rfcs/0000-nixpkgs-ifd.md

```diff
@@ -25,158 +25,171 @@ To this new generation of developers, the distro (or homebrew) is a crufty relic
-   Increasing the critical path length of *everything* we do with Nixpkgs would cause real pain in some quarters, and I don't want that to pay that as the cost of IFD.
+- Similar Bootstrapping problems as the other alternative below: new generated code needs nixpkgs and a previous version of the generated code.
 
-   With the plan as written, users of packages depending on IFD do have to wait slightly longer as the first eval is longer (and we wait for it before beginning to build `allImportedDerivations`).
-   But I think that is fair; we would be the "new constituency", the bottom of the pecking order, and so we should be patient so that other's workflows are not disturbed.
+- `builtins.fetch*` is a nuisance to deal with at the moment and would probably need to be improved to make this work.
+  E.g. gcrooting this evaluation only dependency could prove tricky without changes to Nix.
 
-   Longer term we could revisit this, or we could e.g. double down on automatic vendoring, committing all generated code to a second "roll-up" repo.
-   Many options between those two extremes; I rather not worry to much about it now and just take the conservative polite route proposed here to begin.
+- Extra bureaucracy would be involved with updating the generated repository and the reference to it in nixpkgs.
+  Additionally, special support in CI would be required for this.
 
-2. As always, do nothing, and keep the status quo.
+### Nixpkgs itself becomes a derivation output
 
-# Unresolved questions
-[unresolved]: #unresolved-questions
+This alternative implementation was proposed by @L-as at the meeting.
+The idea is that nixpkgs would become a derivation that builds a “regular” nixpkgs source tree by augmenting files available statically with code generation.
 
-Should we call it "import from Derivation", or should we give it a different name?
-`builtins.readFile <drv>` is really the same thing for our purposes, so I am sympathetic to renaming.
+The upside of this would be that there would only be one instance of IFD that can ever happen, namely when the source tree is built.
+The produced store path then would require no IFD, and it would be obvious what relates to IFD and what doesn't.
 
-# Future work
-[future]: #future-work
+In practice, IFD would not be necessary for users of nixpkgs if we can design a mechanism that allows the dynamically produced nixpkgs source tree to be used as a channel.
+Then the IFD would only need to be executed when working on nixpkgs.
+
+#### Drawbacks
+
+- This approach creates a bootstrapping problem for the entirety of nixpkgs, not just for the IFD parts.
+  It would be necessary to build the new nixpkgs source tree using an old version of the nixpkgs source tree.
+  This could either be done using a fixed “nixpkgs bootstrap tarball” which occasionally needs to be bumped manually as code generation tools require newer dependencies, or by pulling in the latest nixpkgs source tree produced by e.g. Hydra.
```

Comment:
Note that these designs are alternatives to the now proposed “conservative” idea and came up in our initial meeting. They would have the advantage that the generated code would no longer need to be checked into nixpkgs, so I think this is why they are filed under Future Work. Accepting the RFC would not necessarily mean commiting to those designs – best case would be granular IFD where we don't need to check in anything anywhere.

@Ericson2314 Maybe it would be clearer to have them under Alternatives?